src/main/java/com/festy/app/Festy/product/ProductServices.java

```diff
@@ -46,4 +46,20 @@ public Product getProductById(long id){
     public void deleteProduct(Long id){
         repository.deleteById(id);
     }
+
+    public void uppdateProduct(ProductUpdateDTO productUpdateDTO) {
+        Optional< Product > optional = repository.findById(productUpdateDTO.getId());
+        Product product = null;
+        if (optional.isPresent()) {
```

Comment:
Please remove the if statement
and change 
`Product product = null `
to 
`Product product = optional.orElseThrow(() -> new RuntimeException("product not found for id"+productUpdateDTO.getId()))`

Reply:
done